src/Analyzers/CSharp/Analyzers/RemoveUnnecessaryLambdaExpression/CSharpRemoveUnnecessaryLambdaExpressionDiagnosticAnalyzer.cs

```diff
@@ -135,6 +135,15 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol?
         if (invokedMethod.IsPartialDefinition && invokedMethod.PartialImplementationPart is null)
             return;
 
+        // Check if the invoked method is a member of a struct. If it is, and it is not readonly and the invoked method
+        // is not readonly, then we don't want to convert it to a method group. If we did, we may accidently change the
+        // behaviour of the code, since structs are value types and the method group will be copied rather than referenced.
+        if (invokedMethod.ContainingType?.TypeKind == TypeKind.Struct)
```

Comment:
```suggestion
        if (invokedMethod.ContainingType.TypeKind == TypeKind.Struct)
```

Don't need `?.` here  as methods are guaranteed to have containing types.

Reply:
Good catch, fixed.